learning_resources/serializers.py

```diff
@@ -392,8 +423,49 @@ def update(self, instance, validated_data):
 
     class Meta:
         model = models.LearningResource
-        exclude = ["etl_source", *COMMON_IGNORED_FIELDS]
-        read_only_fields = ["platform", "offered_by"]
+        exclude = ["resources", "etl_source", *COMMON_IGNORED_FIELDS]
+        read_only_fields = ["platform", "offered_by", "readable_id"]
+
+
+class PodcastResourceSerializer(LearningResourceBaseSerializer):
+    """Serializer for podcast resources"""
+
+    resource_type = LearningResourceTypeField(
+        default=constants.LearningResourceType.podcast.name
+    )
+
+    podcast = PodcastSerializer(read_only=True)
+
+
+class PodcastEpisodeResourceSerializer(LearningResourceBaseSerializer):
+    """Serializer for podcast episode resources"""
+
+    resource_type = LearningResourceTypeField(
+        default=constants.LearningResourceType.podcast_episode.name
+    )
+
+    podcast_episode = PodcastEpisodeSerializer(read_only=True)
+
+
+class LearningResourceSerializer(serializers.Serializer):
+    """Serializer for LearningResource"""
+
+    serializer_cls_mapping = {
+        serializer_cls().fields["resource_type"].default: serializer_cls
+        for serializer_cls in (
+            ProgramResourceSerializer,
+            CourseResourceSerializer,
+            LearningPathResourceSerializer,
+            PodcastResourceSerializer,
+            PodcastEpisodeResourceSerializer,
+        )
+    }
+
+    def to_representation(self, instance):
+        """Serialize a LearningResource based on resource_type"""
+        serializer_cls = self.serializer_cls_mapping[instance.resource_type]
+
+        return serializer_cls(instance=instance).data
```

Comment:
A few of the learning resource serializers access `self.context` (see `get_learning_path_parents` or `get_user_list_parents`). Maybe we could change that, but for now I think we need to pass `context=self.context` here.


### Resulting bug
You can see the consequences of this if you:
1. Create a learningpath with some items
2. As a staff user, view `localhost:8063/api/v1/learningpaths/<PATH_ID>`
3. The `learning_path_parents` field is `[]`, but should be populated.

**We have a test that should have caught this**...but the test isn't doing what is is supposed to be doing. In `test_get_resource_learning_paths`, 
```python
for idx, item in items_json:
    assert item.get("id") == path_items[idx].id
    assert item.get("position") == path_items[idx].position
    assert item.get("child") == course.learning_resource.id
```
`items_json` and `path_items` are supposed to be the same-ish, but `items_json` is empty on this branch, so test still passes. (Actually it's empty on main, too, since that test doesn't authenticate the user.)

We should add an `assert len(items_json) == 3` or something to that test.